dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/enrollment/DefaultEnrollmentService.java

```diff
@@ -277,59 +286,54 @@ public void decideAccess(EnrollmentQueryParams params) {
-    String violation = null;
-
-    if (params == null) {
-      throw new IllegalQueryException("Params cannot be null");
-    }
-
     if (params.hasProgram() && params.hasTrackedEntityType()) {
-      violation = "Program and tracked entity cannot be specified simultaneously";
+      throw new IllegalQueryException(
+          "Program and tracked entity cannot be specified simultaneously");
     }
 
     if (params.hasProgramStatus() && !params.hasProgram()) {
-      violation = "Program must be defined when program status is defined";
+      throw new IllegalQueryException("Program must be defined when program status is defined");
     }
 
     if (params.hasFollowUp() && !params.hasProgram()) {
-      violation = "Program must be defined when follow up status is defined";
+      throw new IllegalQueryException("Program must be defined when follow up status is defined");
     }
 
     if (params.hasProgramStartDate() && !params.hasProgram()) {
-      violation = "Program must be defined when program start date is specified";
+      throw new IllegalQueryException(
+          "Program must be defined when program start date is specified");
     }
 
     if (params.hasProgramEndDate() && !params.hasProgram()) {
-      violation = "Program must be defined when program end date is specified";
+      throw new IllegalQueryException("Program must be defined when program end date is specified");
     }
 
     if (params.hasLastUpdated() && params.hasLastUpdatedDuration()) {
-      violation = "Last updated and last updated duration cannot be specified simultaneously";
+      throw new IllegalQueryException(
+          "Last updated and last updated duration cannot be specified simultaneously");
     }
 
     if (params.hasLastUpdatedDuration()
         && DateUtils.getDuration(params.getLastUpdatedDuration()) == null) {
-      violation = "Duration is not valid: " + params.getLastUpdatedDuration();
-    }
-
-    if (violation != null) {
-      log.warn("Validation failed: " + violation);
-
-      throw new IllegalQueryException(violation);
+      throw new IllegalQueryException("Duration is not valid: " + params.getLastUpdatedDuration());
     }
   }
 
   private List<Enrollment> getEnrollments(
-      Iterable<Enrollment> enrollments, EnrollmentParams params, boolean includeDeleted)
+      Iterable<Enrollment> enrollments,
+      EnrollmentParams params,
+      boolean includeDeleted,
+      OrganisationUnitSelectionMode orgUnitMode)
       throws ForbiddenException {
     List<Enrollment> enrollmentList = new ArrayList<>();
     User user = currentUserService.getCurrentUser();
 
     for (Enrollment enrollment : enrollments) {
       if (enrollment != null
-          && trackerOwnershipAccessManager.hasAccess(
-              user, enrollment.getTrackedEntity(), enrollment.getProgram())) {
-        enrollmentList.add(getEnrollment(enrollment, params, includeDeleted));
+          && (trackerOwnershipAccessManager.hasAccess(
```

Comment:
Just a small performance fix, if we swap the order here, putting `orgUnitMode == ALL` in the first place, if it is true, then the ownershipCheck, that here is the heavier operation, will not happen